### ISSUE TITLE
Accessibility: Caption for horizontal row widget not associated with the image

### ISSUE THREAD

**Issue by caaster** [projectblacklight/spotlight#3474]
### Issue Description - WCAG standard
See the markup for the image on this exhibit page: https://exhibits.stanford.edu/stanford-aapi/feature/timeline, which shows that the caption is NOT associated with the image. This is incorrect for accessibility.

![Image](https://github.com/user-attachments/assets/1f4699d1-39f4-41a1-8f13-057387141489)

### Method of discovery
Stanford Office of Digital Accessibility review of Exhibits, Stanford's Spotlight instance. 

### Steps to Reproduce
Create content on a home, feature, or about page using the horizontal row widget. Make sure to create a caption.

We should also check all of the widgets that allow captions to see if the same problem is present.

### Suggested solution

![Image](https://github.com/user-attachments/assets/f4407b66-d90d-4c5d-a89a-64179383152a)
